Marketplace/src/main/java/com/marketplace/service/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,92 @@
+package com.marketplace.service.order;
+
+import com.marketplace.config.exception.AddEntryException;
+import com.marketplace.config.exception.ModifyingEntryException;
+import com.marketplace.repository.order.OrderQuery;
+import com.marketplace.repository.order.OrderRepository;
+import com.marketplace.util.OrderMapper;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+public class OrderServiceImpl implements OrderService {
+
+    private final OrderRepository repository;
+    private final OrderMapper mapper;
+
+    public OrderServiceImpl(OrderRepository repository, OrderMapper mapper) {
+        this.repository = repository;
+        this.mapper = mapper;
+    }
+
+
+    @Override
+    public void addOrder(String username) {
+        long userId = repository.getUserId(username);
+
+        if (repository.addOrder(userId) == -1) {
+            throw new AddEntryException(
+                    String.format("new order for user: %s was not added", username)
+            );
+        }
+    }
+
+    @Override
+    public List<OrderedProduct> getCurrentOrder(String username) {
+        long userId = repository.getUserId(username);
+        long orderId = repository.getCurrentOrderId(username);
+
+        return repository.getOrder(userId, orderId)
+                .stream()
+                .map(product -> mapper.toOrderedProduct(product))
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public Map<LocalDate, List<OrderedProduct>> getAllOrders(String username) {
+        long userId = repository.getUserId(username);
+
+        return repository.getAllOrders(userId)
```

Comment:
Use Stream.map() to convert to OrderedProduct first.

Reply:
I cant't do this because i'm grouping objects by field that is not present in mapped objects.